Objects/abstract.c

```diff
@@ -34,6 +34,17 @@ null_error(void)
     return NULL;
 }
 
+static PyObject *
+immutable_error(void)
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    if (!_PyErr_Occurred(tstate)) {
+        _PyErr_SetString(tstate, PyExc_TypeError,
+                         "cannot modify immutable instance");
+    }
+    return NULL;
+}
```

Comment:
Why does this used a custom `immutable_error` function and not `PyErr_WriteToImmutable()`. The difference seems to be that this function doesn't take the object that caused the error. But looking at the `Py_CHECKWRITE(o)` we always seem to have an object